extensions-contrib/opentelemetry-extensions/src/test/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryBenchmark.java

```diff
@@ -0,0 +1,146 @@
+
+  private static final InputRowSchema ROW_SCHEMA = new InputRowSchema(null,
+      new DimensionsSpec(ImmutableList.of(
+          new StringDimensionSchema("name"),
+          new StringDimensionSchema("value"),
+          new StringDimensionSchema("foo_key")),
+          null, null),
+      null);
+
+  private static final OpenTelemetryMetricsProtobufInputFormat INPUT_FORMAT =
+      new OpenTelemetryMetricsProtobufInputFormat("name",
+          "value",
+          "",
+          "resource.");
+
+  private static ByteBuffer createMetricBuffer(Integer resourceMetricCount,
+                                               Integer instrumentationLibraryCount,
+                                               Integer metricsCount,
+                                               Integer dataPointCount)
+  {
+    MetricsData.Builder metricsData = MetricsData.newBuilder();
+    for (int i = 0; i < resourceMetricCount; i++) {
+      ResourceMetrics.Builder resourceMetricsBuilder = metricsData.addResourceMetricsBuilder();
+      Resource.Builder resourceBuilder = resourceMetricsBuilder.getResourceBuilder();
+
+      for (int resourceAttributeI = 0; resourceAttributeI < 5; resourceAttributeI++) {
+        KeyValue.Builder resourceAttributeBuilder = resourceBuilder.addAttributesBuilder();
+        resourceAttributeBuilder.setKey("resource.label_key_" + resourceAttributeI);
+        resourceAttributeBuilder.setValue(AnyValue.newBuilder().setStringValue("resource.label_value"));
+      }
+
+      for (int j = 0; j < instrumentationLibraryCount; j++) {
+        InstrumentationLibraryMetrics.Builder instrumentationLibraryMetricsBuilder =
+            resourceMetricsBuilder.addInstrumentationLibraryMetricsBuilder();
+
+        for (int k = 0; k < metricsCount; k++) {
+          Metric.Builder metricBuilder = instrumentationLibraryMetricsBuilder.addMetricsBuilder();
+          metricBuilder.setName("io.confluent.domain/such/good/metric/wow");
+
+          for (int l = 0; l < dataPointCount; l++) {
+            NumberDataPoint.Builder dataPointBuilder = metricBuilder.getSumBuilder().addDataPointsBuilder();
+            dataPointBuilder.setAsDouble(42.0).setTimeUnixNano(TIMESTAMP);
+
+            for (int metricAttributeI = 0; metricAttributeI < 10; metricAttributeI++) {
+              KeyValue.Builder attributeBuilder = dataPointBuilder.addAttributesBuilder();
+              attributeBuilder.setKey("foo_key_" + metricAttributeI);
+              attributeBuilder.setValue(AnyValue.newBuilder().setStringValue("foo-value"));
+            }
+          }
+        }
+      }
+    }
+    return ByteBuffer.wrap(metricsData.build().toByteArray());
+  }
+
+  @Benchmark()
+  public void measureSerde(Blackhole blackhole) throws IOException
+  {
+    for (CloseableIterator<InputRow> it = INPUT_FORMAT.createReader(ROW_SCHEMA, new ByteEntity(SIMPLE_BUFFER), null).read(); it.hasNext(); ) {
```

Comment:
the buffer needs to be duplicated every time it is passed to the reader, otherwise we only de-serialize the first time and subsequent iterations are just reading an empty array, since the buffer position would have been changed.

Reply:
The ByteEntity constructor already [has a call](https://github.com/confluentinc/druid/blob/0.21.0-confluent/core/src/main/java/org/apache/druid/data/input/impl/ByteEntity.java#L36) to `ByteBuffer::duplicate()`